base/src/main/java/org/aya/tyck/pat/PatTycker.java

```diff
@@ -202,7 +194,18 @@ private void addPatSubst(@NotNull Var var, @NotNull Pat pat, @NotNull SourcePos
       }
       case Pattern.CalmFace face -> new Pat.Meta(face.explicit(), new Ref<>(),
         new LocalVar(Constants.ANONYMOUS_PREFIX, face.sourcePos()), term);
-      case default -> throw new UnsupportedOperationException("Number patterns are unsupported yet");
+      case Pattern.Number num -> {
```

Comment:
This is totally wrong. The type has to be interval to make `0`, `1` patterns.

Reply:
Your code makes it impossible to have `0`, `1` patterns work for nat in the future

Reply:
Also ![image](https://user-images.githubusercontent.com/16398479/164559406-4c90c8a5-44d9-41db-9b2b-15538499efca.png)